src/pymor/operators/constructions.py

```diff
@@ -1321,6 +1325,61 @@ def as_source_array(self, mu=None):
         return self.operators[operator_number].as_source_array(mu=mu)
 
 
+def vector_array_to_selection_operator(array, time_instances=None, initial_time=None, end_time=None, name=None):
+    """Create a |SelectionOperator| from a |VectorArray|.
+
+    A |SelectionOperator| is created that selects an element from the |VectorArray| `array`
+    depending on the time component of the parameter that is passed to the |Operator|.
+    If `time_instances` is provided, these values are used for selection according to the
+    behavior of the |SelectionOperator|.
+    If `initial_time` and `end_time` are used, the time interval is split equidistantly where
+    the number of subintervals is determined by the length of `array`.
+    In time, the operator is therefore piecewise constant and the discontinuity points are
+    at `t_n + eps`, i.e.::
+
+        -infty ------- initial_time ----------- t_1 ----------- t_2 ----------- ... ----------- t_{n-1} ----------- end_time ------------------------ infty
+                                     |               |               |                                   |                    |
+        ------------ array[0] -------|--- array[1] --|--- array[2] --|--------- ... ---------------------|---- array[n-2] ----|-------- array[n-1] --------
+                                     |               |               |                                   |                    |
+
+    where `n = len(array)`, `t_i = initial_time + i * (end_time - initial_time) / n`,
+    `eps = (end_time - initial_time) * machine_eps * n * 10` and
+    the discontinuity points are of the form
+    `initial_time + j * (end_time - initial_time) / n + eps` for `j = 0,...,n-1`.
```

Comment:
This part can be removed since you already state above that they are of the form `t_i + eps`.